### PR TITLE
Optimizes round robin traffic class and adds a round robin benchmark.

### PR DESCRIPTION
**Pull request by barath**

### CODE REVIEW

File: core/traffic_class.cc [hunk 1]
```diff
@@ -200,15 +198,17 @@ void RoundRobinTrafficClass::FinishAndAccountTowardsRoot(
...
-    if (children_.empty()) {
-      blocked_ = true;
-    }
   } else {
-    children_.push_back(child);
+    ++next_child_;
+  }
+
+  // Wrap around for round robin.
+  if (next_child_ >= children_.size()) {
```
Comment (melvinw) [inline core/traffic_class.cc]: is this faster than using `%`?

**Comment by melvinw** [comment]
looks good! just one small comment

File: core/traffic_class.cc (hunk 1 above)
Comment (barath) [inline core/traffic_class.cc]: I believe so, but I didn't benchmark it.  Let me do so and will report back.

**Comment by melvinw** [comment]
thanks! 
cat

**Pull request merged**